### PR TITLE
Add support for Debian 11 and Ubuntu 22.04 platforms

### PR DESCRIPTION
**Pull request by narph**
## What does this PR do?

Adds support for Debian 11 and Ubuntu 22.04 platforms 

## Why is it important?

Adds support for Debian 11 and Ubuntu 22.04 platforms.
Latest Debian and Ubuntu versions are already supported for other stack components and customers are asking if we support it as well. 
We need to make sure elastic agent/fleet server tests will pass on these 2 platforms as well.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/235

### CODE REVIEW

File: .ci/.e2e-platforms.yaml [hunk 1]
```diff
@@ -40,3 +40,13 @@ PLATFORMS:
     image: "ami-0d90bed76900e679a"
     instance_type: "t3.xlarge"
     username: admin
+  debian_11_amd64:
+    description: "Debian 11 AMD64"
+    image: "ami-0c6acff4e66eba1f1"
+    instance_type: "t3.xlarge"
+    username: admin
+  ubuntu_22.04:
+    description: "Ubuntu 22.04"
```
Comment (mdelapenya) [inline .ci/.e2e-platforms.yaml:49]:
```suggestion
    description: "Ubuntu 22.04 AMD"
```

**Review by mdelapenya — CHANGES_REQUESTED** [review]
This looks good to me, although we need one thing more, as described in https://github.com/elastic/e2e-testing/pull/2589/files#diff-53b78ab29e8f8423188da6c0806aaa9de90d39bd3502d83a655a1a68429fcfe2R43

File: .ci/.e2e-platforms.yaml [hunk 2]
```diff
@@ -40,3 +40,13 @@ PLATFORMS:
     image: "ami-0d90bed76900e679a"
     instance_type: "t3.xlarge"
     username: admin
+  debian_11_amd64:
+    description: "Debian 11 AMD64"
+    image: "ami-0c6acff4e66eba1f1"
+    instance_type: "t3.xlarge"
+    username: admin
+  ubuntu_22.04:
```
Comment (mdelapenya) [inline .ci/.e2e-platforms.yaml:48]: I'd say `ubuntu_22.04` is not a good YAML key:
```suggestion
  ubuntu_22_04_amd:
```

Comment (narph) [reply .ci/.e2e-platforms.yaml:48]: made the changes

File: .ci/.e2e-platforms.yaml [hunk 3]
```diff
@@ -40,3 +40,13 @@ PLATFORMS:
     image: "ami-0d90bed76900e679a"
     instance_type: "t3.xlarge"
     username: admin
+  debian_11_amd64:
```
[the remark this answers is not in the document]
Comment (narph) [reply .ci/.e2e-platforms.yaml:43]: renamed debian

[the remark this answers is not in the document]
Comment (mdelapenya) [reply .ci/.e2e-platforms.yaml:43]: Great! Could you please update the different profile files, which are used by beats and the elastic-agent builds?
- .ci/.e2e-test.yaml
- .ci/.e2e-test-beats.yaml
- .ci/.e2e-test-for-elastic-agent.yaml
- .ci/.e2e-test-daily.yaml
🙏

[the remark this answers is not in the document]
Comment (mdelapenya) [reply .ci/.e2e-platforms.yaml:43]: I updated the docs PR: https://github.com/elastic/e2e-testing/pull/2589/files#r889015917

File: .ci/.e2e-tests-daily.yaml [hunk 4]
```diff
@@ -5,51 +5,51 @@ SUITES:
     scenarios:
```
Comment (mdelapenya) [inline .ci/.e2e-tests-daily.yaml:5]: I think we want to include debian11 and ubuntu in the nightly builds. So I guess they should be added here too.

@jlind23 could you confirm?

File: .ci/.e2e-platforms.yaml [hunk 5]
```diff
@@ -40,3 +40,13 @@ PLATFORMS:
...
+    description: "Debian 11 AMD64"
+    image: "ami-0c6acff4e66eba1f1"
+    instance_type: "t3.xlarge"
+    username: admin
+  ubuntu_22.04_amd64:
```
Comment (mdelapenya) [inline .ci/.e2e-platforms.yaml:48]: Is this YAML key valid, including a `.`? I recommend changing it to `ubuntu_22_04_amd64`

**Comment by mdelapenya** [comment]
For Debian, I'm checking their website for the AMIs: https://wiki.debian.org/Cloud/AmazonEC2Image/Bullseye, where they say that `ami-0c7c4e3c6b4941f0f` is the AMI in us-east-2.

I do not know why I'm not able to discover that AMI ID in the AWS console 🤷  Anyway. I checked locally and I'm able to create a node with that AMI, successfully running the test scenarios on it 🚀

**Comment by mdelapenya** [comment]
Ok, this is working now. There are two failures that I'd recommend to investigate before merging, as they affect Debian 11:

```shell
make -C .ci setup-stack  # it will fail as the fleet-server needs a different healthcheck for Kibana + Fleet
make -C .ci recreate-fleet-server # causes the stack to restart fleet-server
make -C .ci provision-node NODE_IMAGE=ami-0c7c4e3c6b4941f0f NODE_LABEL=debian_11_amd64
make -C .ci setup-node NODE_IMAGE=ami-0c7c4e3c6b4941f0f NODE_LABEL=debian_11_amd64
make -C .ci run-tests TAGS="backend_processes" NODE_LABEL=debian_11_amd64
make -C .ci run-tests TAGS="system_integration" NODE_LABEL=debian_11_amd64
```

**Comment by narph** [comment]
> Ok, this is working now. There are two failures that I'd recommend to investigate before merging, as they affect Debian 11:
> 
> ```shell
> make -C .ci setup-stack  # it will fail as the fleet-server needs a different healthcheck for Kibana + Fleet
> make -C .ci recreate-fleet-server # causes the stack to restart fleet-server
> make -C .ci provision-node NODE_IMAGE=ami-0c7c4e3c6b4941f0f NODE_LABEL=debian_11_amd64
> make -C .ci setup-node NODE_IMAGE=ami-0c7c4e3c6b4941f0f NODE_LABEL=debian_11_amd64
> make -C .ci run-tests TAGS="backend_processes" NODE_LABEL=debian_11_amd64
> make -C .ci run-tests TAGS="system_integration" NODE_LABEL=debian_11_amd64
> ```

I see there are the same flaky tests, the Debian 11 tests just passed. Can you confirm?

**Comment by mdelapenya** [comment]
Yes, it's seems the debian/Ubuntu support is at the same level than the rest of platforms.


LGTM

**Review by mdelapenya — APPROVED** [review]
Ready!! 🚀

File: .ci/.e2e-tests-beats.yaml [hunk 6]
```diff
@@ -5,39 +5,39 @@ SUITES:
...
       - name: "Metricbeat"
         tags: "metricbeat"
-        platforms: ["debian_amd64"]
+        platforms: ["debian_10_amd64"]
   - suite: "fleet"
     scenarios:
       - name: "Fleet"
         tags: "fleet_mode_agent"
-        platforms: ["centos8_arm64", "centos8_amd64", "debian_arm64", "debian_amd64", "sles15"]
+        platforms: ["centos8_arm64", "centos8_amd64", " debian_10_arm64", "debian_10_amd64", "sles15"]
```
Comment (mdelapenya) [inline .ci/.e2e-tests-beats.yaml:16]: @v1v we found it! 🤦

**Pull request merged**